File: jclang/src/main/java/jclang/IndexerCallback.java


package jclang;

import org.jetbrains.annotations.NotNull;

import java.io.File;

public interface IndexerCallback {
    IndexerCallback DO_NOTHING = new AbstractIndexerCallback() {};

    void enteredMainFile(@NotNull File mainFile);

    void startedTranslationUnit();

    void indexDeclaration(@NotNull DeclarationInfo info);
}

File: libclang/src/main/java/klang/jvm/IndexerCallback.kt
package klang.jvm

import java.io.File

interface IndexerCallback {
    fun enteredMainFile(mainFile: File)
    fun startedTranslationUnit()
    fun indexDeclaration(info: DeclarationInfo)

    companion object {
        val DO_NOTHING: IndexerCallback = object : AbstractIndexerCallback() {}
    }
}